### PR TITLE
Add PSDocs automated workflow

### PR DESCRIPTION
**Pull request by BernieWhite**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Added documentation automation with PSDocs for Azure.

### CODE REVIEW

**Comment by BernieWhite** [comment]
@bmoore-msft Initial thoughts?

File: .ps-docs/azure-template-badges.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+![Azure Public Test Date](https://azurequickstartsservice.blob.core.windows.net/badges/{{ template_path }}/PublicLastTestDate.svg)
+![Azure Public Test Result](https://azurequickstartsservice.blob.core.windows.net/badges/{{ template_path }}/PublicDeployment.svg)
+
+![Azure US Gov Last Test Date](https://azurequickstartsservice.blob.core.windows.net/badges/{{ template_path }}/FairfaxLastTestDate.svg)
+![Azure US Gov Last Test Result](https://azurequickstartsservice.blob.core.windows.net/badges/{{ template_path }}/FairfaxDeployment.svg)
+
+![Best Practice Check](https://azurequickstartsservice.blob.core.windows.net/badges/{{ template_path }}/BestPracticeResult.svg)
+![Cred Scan Check](https://azurequickstartsservice.blob.core.windows.net/badges/{{ template_path }}/CredScanResult.svg)
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F{{ template_path_encoded }}%2Fazuredeploy.json)
```
Comment (bmoore-msft) [inline .ps-docs/azure-template-badges.md:10]: For each button, we need to make sure the buttons match the cloud support (i.e. the environments property) in metadata.json.  See the Validate-Metadata.ps1 script for slightly more info...

Comment (BernieWhite) [reply .ps-docs/azure-template-badges.md:10]: @bmoore-msft Do we care about the badges or just the deployment buttons?

Comment (bmoore-msft) [reply .ps-docs/azure-template-badges.md:10]: Both... buttons should match badges... e.g. if USGov is "not supported" there should be no button for it...

**Pull request closed without merging**